sph/test/kernel_ve/density_kern.cpp

```diff
@@ -88,12 +86,10 @@ TEST(Density, JLoop)
                                wh.data(),
                                whd.data(),
                                xm.data(),
-                               wrho0.data(),
-                               rho.data(),
                                kx.data(),
                                gradh.data());
-    EXPECT_NEAR(rho[0], 1.67849454056818e-2, 1e-10);
-    EXPECT_NEAR(gradh[0], 1.2501347388453987, 1e-10);
```

Comment:
The gradh reference had to be changed because the input whro0 was inconsistent with the whro0 that is computed on the fly.
Before changing it, I added a runtime check that the previous whro0 computed in Xmass is identical to the whro0 computed on-the-fly in density.